server/controllers/auth/signup.ts

```diff
@@ -0,0 +1,36 @@
+import { Response, Request, NextFunction } from 'express';
+import { hash } from 'bcrypt';
+import { createNewUserQuery, checkEmailTakenQuery } from '../../database/queries';
+import { signupSchema, signToken } from '../../utils';
+import { CustomError } from '../../errors';
+
+export default async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const {
+      role, email, username, password, bio, avatar,
+    } = req.body;
+
+    await signupSchema.validate({ ...req.body, role });
```

Comment:
Why you send role alone from the body, even though it's part of it?

Reply:
Oooh, you're right 👍